ckan/lib/app_globals.py

```diff
@@ -221,7 +221,7 @@ def _init(self):
         self.ckan_version = ckan.__version__
         self.ckan_base_version = re.sub(r'[^0-9\.]', '', self.ckan_version)
         if self.ckan_base_version == self.ckan_version:
-            self.ckan_doc_version = self.ckan_version[:3]
+            self.ckan_doc_version = (".".join(self.ckan_version.split(".")[:2]), ".".join(self.ckan_version.split(".")[2:]))[0]
         else:
             self.ckan_doc_version = 'latest'
```

Comment:
This would work but to be honest we should use a proper package to parse and compare versions everywhere. Let's start using [`packaging.version.parse`](https://packaging.pypa.io/en/latest/version.html), which makes things much more robust and easier to read.

```python
from packaging.version import parse as parse_version

        self.ckan_version = ckan.__version__
        version = parse_version(self.ckan_version)
        self.ckan_base_version = vesion.base_version
        if not version.is_prerelease:
            self.ckan_doc_version = f"{version.major}.{version.minor}"
        else:
            self.ckan_doc_version = "latest"
```

What do you think @Mikanebu ?

Reply:
@amercader Sounds good!

Reply:
@amercader Do you want me to update PR with packaging?

Reply:
Sorry @Mikanebu I missed this, yes an update to use packaging would be great